HTML/Miscellaneous.tmPreferences

```diff
@@ -9,29 +9,25 @@
 	<dict>
 		<key>decreaseIndentPattern</key>
 		<string><![CDATA[(?x)
-			^\s*
-			<\?(php)?\s+(else(if)?|end(if|for(each)?|while))
-		]]></string>
-		<key>batchDecreaseIndentPattern</key>
-		<string><![CDATA[(?x)
-			^\s*
-			(</(?!html)
-			  [A-Za-z0-9-]+\b[^>]*>
-			|-->
-			|<\?(php)?\s+(else(if)?|end(if|for(each)?|while))
-			|\}
+			^\s*                                                                              # the beginning of the line followed by any amount of whitespace
+			(
+			    <\?(php)?\s+(else(if)?|end(if|for(each)?|while))                              # ending php control keywords
+			|   </(?!html)[A-Za-z0-9-]+\b[^>]*>                                               # any valid close tag except html
+			|   -->                                                                           # closing comment punctuation
+			|   \}                                                                            # a closing brace
 			)
 		]]></string>
 		<key>increaseIndentPattern</key>
-		<string><![CDATA[^\s*<\?php\s*\b(if|else|elseif|for|foreach|while)\b.*:(?!.*end\1)]]></string>
-		<key>batchIncreaseIndentPattern</key>
 		<string><![CDATA[(?x)
-			^\s*
-			<(?!\?|area|base|br|col|frame|hr|html|img|input|link|meta|param|[^>]*/>)
-			  ([A-Za-z0-9-]+)(?=\s|>)\b[^>]*>(?!.*</\1>)
-			|<!--(?!.*-->)
-			|<\?php.+?\b(if|else(?:if)?|for(?:each)?|while)\b.*:(?!.*end\1)
-			|\{[^}"']*$
+			^\s*                                                                              # the beginning of the line followed by any amount of whitespace
+			(
+			    <\?(php)?\s*\b(?<php_control_word>if|else|elseif|for|foreach|while)\b.*:      # php control keywords
+			    (?!.*end\g<php_control_word>)                                                 # - that don't end themselves on the same line
```

Comment:
I know it's been like this before, but using a backref here is quite naive because `elseif` is closed by `endif` and not `endelseif` (similar for `else`, `foreach`). One could batch the keywords that share the same end keyword in a named group and then resort to conditional expressions for the end block.